packages/realm/src/Realm.ts

```diff
@@ -107,20 +165,54 @@ type InternalConfig = {
 };
 
 export class Realm {
-  public static Object = RealmObject;
+  /** @deprecated Please use named imports */
+  public static App = App;
+  /** @deprecated Please use named imports */
+  public static Auth = { EmailPasswordAuth, ApiKeyAuth };
+  /** @deprecated Please use named imports */
+  public static BSON = BSON;
+  /** @deprecated Please use named imports */
+  public static ClientResetMode = ClientResetMode;
+  /** @deprecated Please use named imports */
   public static Collection = Collection;
+  /** @deprecated Please use named imports */
+  public static ConnectionState = ConnectionState;
+  /** @deprecated Please use named imports */
+  public static Dictionary = Dictionary;
+  /** @deprecated Please use named imports */
+  public static List = List;
+  // TODO: Decide if we want to deprecate this as well
+  public static Object = RealmObject;
+  /** @deprecated Please use named imports */
   public static OrderedCollection = OrderedCollection;
+  /** @deprecated Please use named imports */
+  public static OpenRealmBehaviorType = OpenRealmBehaviorType;
+  /** @deprecated Please use named imports */
+  public static OpenRealmTimeOutBehavior = OpenRealmTimeOutBehavior;
+  /** @deprecated Please use named imports */
+  public static ProgressDirection = ProgressDirection;
+  /** @deprecated Please use named imports */
+  public static ProgressMode = ProgressMode;
+  /** @deprecated Please use named imports */
+  public static ProviderType = ProviderType;
+  /** @deprecated Please use named imports */
   public static Results = Results;
-  public static List = List;
-  public static Dictionary = Dictionary;
+  /** @deprecated Please use named imports */
+  public static SessionState = SessionState;
+  /** @deprecated Please use named imports */
+  public static SessionStopPolicy = SessionStopPolicy;
+  /** @deprecated Please use named imports */
   public static Set = RealmSet;
-  public static App = App;
-  public static UpdateMode = UpdateMode;
-  public static BSON = BSON;
+  /** @deprecated Please use named imports */
+  public static SyncError = SyncError;
+  /** @deprecated Please use named imports */
   public static Types = Types;
+  /** @deprecated Please use named imports */
+  public static UpdateMode = UpdateMode;
+  /** @deprecated Please use named imports */
   public static User = User;
-  public static Credentials = Credentials;
```

Comment:
Great catch. Because of https://github.com/realm/realm-js/tree/bindgen/types#L266-L269 I thought they were exported on `App` only, but looking more closely, this should definitely not be removed.